opencog/combo/type_checker/type_tree.cc

```diff
@@ -1353,7 +1359,19 @@ type_tree gen_signature(const type_tree& itype, const type_tree& otype,
     res.replace(sib, otype.begin());
     return res;
 }
-
+type_node_seq input_out;
+type_node_seq gen_signature(const Handle pg_type){
```

Comment:
Very minor, but the standard requires a space between `)` and `{`.

Reply:
I think we should remove `gen_signature` and `gen_type_node` altogether. If I am not mistaken they aren't used anywhere and I don't think we should have such a thing. gen_signature should return a signature which could be a fully featured atomese signature or a temporary arrow link like `infer_atomese_type` and we might want to implement `type_tree_to_tyn_seq` that takes atomese type signature and return HandleSeq. But for now we don't need it since our `populate` code is using combo type_tree anyway.

Reply:
@kasimebrahim 
We had discussed the idea of having gen_signature to produce an atomese signature but this   output result was picked I believe by you. It will be more useful if we had it with atomese output but I believe this might be useful for future usages as well since we have no other such methods for atomese except `infer_atomese_type` which returns an Arrow_link.

Reply:
Eman, what I suggested was to have a gen_signature that returns a HandleSeq **if we rejected ArrowLink and don't have  a good representation for atomese type signature** and **if we want to use atomese types in populate and other places**. And that was why I also suggested to change a few things in the ITable. But now your gen_signature method expects a type signature in ArrowLink, It is returning a vector of combo types instead of a vector of atomese types plus you are using combo types in populate and we already have working utilities for that isn't that right?

>  I believe this might be useful for future usages

do you have any usecase in mind?

Reply:
I don't have any at the top of my head right now but I remember writing codes in the past where such representation would have been useful. for instance if we want to know the data_typs of the inner arguments. But I guess if the  `infer_atomese_type` can do it then it might be redundant. we can remove it